stellar-contract-env-common/src/raw_val.rs

```diff
@@ -61,7 +61,18 @@ impl AsMut<RawVal> for RawVal {
 // away, the same way `()` would, while remaining a separate type to allow
 // conversion to a more-structured error code at a higher level.
 #[derive(Debug)]
-pub struct ConversionError;
+pub struct ConversionError<F, T>;
```

Comment:
In my suggestion below what I mean is make this as follows:

```suggestion
pub struct ConversionError<T> {
    #[cfg(not(target_family = "wasm"))]
    /// Source is the RawVal that was attempted to be converted into T, but that failed conversion.
    pub source: RawVal;
}
```

Reply:
@leighmcculloch I had to remove the `#[cfg(not(target_family = "wasm"))]` guard because the `ConversionError` being constructed and used in multiple places and I can't find an elegant way to toggle it without putting this guard everywhere. 
> The reason to put all this behind a not-wasm cfg is so that when testing locally this nice error is outputted, but then on chain all this code disappears.

Is it a requirement for this to not show up on-chain?

Reply:
When you introduce the cfg, what size difference do the wasm files have? That's the only thing we can easily measure right at this moment.

I'm inclined to say if there's no substantial size difference run with it without the cfg, and create an issue for evaluating the execution cost in September.

Reply:
I see, then I agree it is important for it to not show up wasm-build.